src/duckling/selection/copy-paste.service.ts

```diff
@@ -22,47 +22,65 @@ export class CopyPasteService {
         this._store.state.subscribe(state => this.clipboard.next(state.clipboard));
     }
 
-    copy(entityKey : EntityKey) {
-        let entity = this._entitySystem.getEntity(entityKey);
-        this._store.dispatch(_copyAction(entity));
+    copy(entities : Entity[]) {
+        this._store.dispatch(_copyAction(entities));
     }
 
-    paste(position : Vector) : EntityKey {
-        let clipboardEntity = this.clipboard.value.copiedEntity;
-        if (clipboardEntity) {
-            return this._entitySystem.addNewEntity(this._position.setPosition(clipboardEntity, vectorRound(position)));
-        } else {
+    paste(position : Vector) : EntityKey[] {
+        let clipboardEntities = this.clipboard.value.copiedEntities;
+        if (!clipboardEntities) {
             return null;
         }
+
+        let pastedEntities: EntityKey[] = [];
+        let lowestPoint = this.findLowestPoint(clipboardEntities);
+        let mergeKey = newMergeKey();
+        for (let entity of clipboardEntities) {
+            let newEntity = this._position.setPosition(entity, vectorRound(vectorAdd(position, vectorSubtract(this._position.getPosition(entity), lowestPoint))));
+            pastedEntities.push(this._entitySystem.addNewEntity(newEntity, mergeKey));
+        }
+        return pastedEntities;
+    }
+
+    findLowestPoint(entities : Entity[]) : Vector {
```

Comment:
Eh, I prefer Tucker's naming here since it doesn't do anything related to the passed in entities being part of a selection. So it seems unnecessary to tie the function to this specific case.